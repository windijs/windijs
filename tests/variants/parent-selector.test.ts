import { backgroundClipConfig, backgroundColor, bundle, colors, configHandler, createUtility, groupActive, groupFocus, groupHover, groupVisited, prop } from "index";

const bg = createUtility("bg")
  .use(backgroundColor(colors))
  .case("clip", configHandler(backgroundClipConfig, ["backgroundClip", prop`-webkit-background-clip`]))
  .init();

const utilities = [bg.blue[500], bg.clip.content];

test("groupHover", () => {
  expect(bundle(groupHover(...utilities))).toMatchSnapshot();
});

test("groupFocus", () => {
  expect(bundle(groupFocus(...utilities))).toMatchSnapshot();
});

test("groupActive", () => {
  expect(bundle(groupActive(...utilities))).toMatchSnapshot();
});

test("groupVisited", () => {
  expect(bundle(groupVisited(...utilities))).toMatchSnapshot();
});
