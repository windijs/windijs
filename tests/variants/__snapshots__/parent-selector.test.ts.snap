// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`groupActive 1`] = `
Object {
  ".group:active &": Object {
    "--w-bg-opacity": "1",
    "-webkit-background-clip": "content-box",
    "backgroundClip": "content-box",
    "backgroundColor": "rgba(59, 130, 246, var(--w-bg-opacity))",
  },
}
`;

exports[`groupFocus 1`] = `
Object {
  ".group:focus &": Object {
    "--w-bg-opacity": "1",
    "-webkit-background-clip": "content-box",
    "backgroundClip": "content-box",
    "backgroundColor": "rgba(59, 130, 246, var(--w-bg-opacity))",
  },
}
`;

exports[`groupHover 1`] = `
Object {
  ".group:hover &": Object {
    "--w-bg-opacity": "1",
    "-webkit-background-clip": "content-box",
    "backgroundClip": "content-box",
    "backgroundColor": "rgba(59, 130, 246, var(--w-bg-opacity))",
  },
}
`;

exports[`groupVisited 1`] = `
Object {
  ".group:visited &": Object {
    "--w-bg-opacity": "1",
    "-webkit-background-clip": "content-box",
    "backgroundClip": "content-box",
    "backgroundColor": "rgba(59, 130, 246, var(--w-bg-opacity))",
  },
}
`;
